### PR TITLE
lint target and workflow job

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up #3634

adds lint target with its workflow job and cleanup old check behavior.

### CODE REVIEW

**Review by milosgajdos — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -48,10 +48,6 @@ version/version.go:
 	@echo "$(WHALE) $@"
 	./version/version.sh > $@
 
-check: ## run all linters (TODO: enable "unused", "varcheck", "ineffassign", "unconvert", "staticheck", "goimports", "structcheck")
```
Comment (thaJeztah) [inline Makefile:51]: do we need to keep `check` as an alias? Or are we ok with removing it? (not sure if there's people expecting `make check` to be there)

Comment (milosgajdos) [reply Makefile:51]: I dont think it's ever been used out of the GHA workflow context.

Comment (thaJeztah) [reply Makefile:51]: alrighty; works for me!

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**